pages/src/hooks/preventParentDeletion.ts

```diff
@@ -0,0 +1,51 @@
+import { CollectionBeforeDeleteHook } from 'payload'
+import { asPageCollectionConfigOrThrow } from '../collections/PageCollectionConfig.js'
+import { PagesPluginConfig } from '../types/PagesPluginConfig.js'
+import { childDocumentsOf } from '../utils/childDocumentsOf.js'
+
+const ADAPTERS_REQUIRING_CUSTOM_LOGIC = [
+  '@payloadcms/db-mongodb',
+  '@payloadcms/db-sqlite',
+  '@payloadcms/db-postgres'
+]
+
+export const preventParentDeletion: CollectionBeforeDeleteHook = async ({
+  req,
+  id,
+  collection,
+}) => {
+  const databaseAdapter = req.payload.db.packageName || req.payload.db.name
+  if (!ADAPTERS_REQUIRING_CUSTOM_LOGIC.includes(databaseAdapter)) {
+    return
+  }
+
+  const pageConfig = asPageCollectionConfigOrThrow(collection)
+  const pagesPluginConfig = collection.custom?.pagesPluginConfig as PagesPluginConfig
+  
+  const childDocuments = await childDocumentsOf(
+    req,
+    id,
+    collection.slug,
+    pagesPluginConfig?.baseFilter
+  )
+
+  if (childDocuments.length > 0) {
+    const childrenByCollection = childDocuments.reduce((acc, child) => {
+      if (!acc[child.collection]) {
+        acc[child.collection] = []
+      }
+      acc[child.collection].push(child.id)
+      return acc
+    }, {} as Record<string, (string | number)[]>)
+
+    const collectionMessages = Object.entries(childrenByCollection)
+      .map(([collectionSlug, ids]) => 
+        `${ids.length} document(s) in the "${collectionSlug}" collection`
+      )
+      .join(', ')
+
+    const errorMessage = `Cannot delete this document because it is referenced as a parent by ${collectionMessages}. Please remove or reassign the child documents before deleting this parent document.`
+    
+    throw new Error(errorMessage)
```

Comment:
While testing, I noticed that the message is not shown in the admin panel. Instead it says 'Something went wrong' in the toast, which is bad UX. Please use the `AdminPanelError` class from`utils/adminPanelError` to ensure the message is visible to the user.